user_guide.adoc

```diff
@@ -1090,7 +1090,117 @@ bytes sparsely.
 
 === Streams and substreams
 
-TODO
+====Introduction and simple example====
+
+A stream is a flow of data from an input file into a parser which is
+generated by a KS script. The parser can request one or more bits of
+data from the stream at a time, but cannot request the same data twice
+and cannot request data be provided out of sequential order. A stream
+knows the maximum amount of data available to be requested by the
+parser and the actual amount of data which has already been
+requested by the parser.
```

Comment:
This explanation is pretty abstract and somewhat misleading. "Stream" can be re-read as many times as needed, and it can be seeked: that's exactly how positional parse instances work, they use `seek` operations on a stream.